src/main/java/kg/jarkyn/server/ReceivingSocket.java

```diff
@@ -0,0 +1,22 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+public class ReceivingSocket {
+    private ServerSocket serverSocket;
+
+    public ReceivingSocket(ServerSocket serverSocket) {
```

Comment:
So at the moment ReceivingSocket returns a RequestingSocket which returns an inputStream which is given to requestParser which then returns the parsedRequest.

I think there is room for simplifying this/removing unnecessary steps. For e.g. couldn't this ReceivingSocket just take an opened Socket and return the parsedRequest.